packages/angular/ssr/test/app_spec.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+/* eslint-disable import/no-unassigned-import */
+import 'zone.js/node';
+import '@angular/compiler';
+/* eslint-enable import/no-unassigned-import */
+
+import { AngularServerApp } from '../src/app';
+import { ServerRenderContext } from '../src/render';
+import { setAngularAppTestingManifest } from './testing-utils';
+
+describe('AngularServerApp', () => {
+  let app: AngularServerApp;
+
+  beforeAll(() => {
+    setAngularAppTestingManifest();
+    app = new AngularServerApp({
+      isDevMode: true,
+    });
+  });
+
+  describe('render', () => {
+    it(`should include 'ng-server-context="ssr"' by default`, async () => {
+      const response = await app.render(new Request('http://localhost/home'));
+      expect(await response.text()).toContain('ng-server-context="ssr"');
+    });
+
+    it(`should include the provided 'ng-server-context' value`, async () => {
+      const response = await app.render(
+        new Request('http://localhost/home'),
+        undefined,
+        ServerRenderContext.SSG,
+      );
+      expect(await response.text()).toContain('ng-server-context="ssg"');
+    });
+
+    it('should correctly render the content for the requested page', async () => {
+      const response = await app.render(new Request('http://localhost/home'));
+      expect(await response.text()).toContain('Home works');
```

Comment:
Consider: All these tests check for `/home` and `"Home works"` but exactly where those strings come from is kind of far away in `./testing-utils.ts`. What if we moved the routes object as an input to `setAngularAppTestingManifest` and provided them in this file so that actual data being asserted is properly localized to this test?

```typescript
@Component({
        standalone: true,
        selector: 'app-home',
        template: `Home works ${locale.toUpperCase()}`,
})
class HomeComponent {}

beforeAll(() => {
  setAngularAppTestingManifest([
    {path: 'home', component: HomeComponent},
  ]);
  app = new AngularServerApp({
    isDevMode: true,
  });
});
```